### PR TITLE
Add more exception handling for portable PDB writing in PeWriter

### PR DESCRIPTION
**Pull request by agocke**
**Customer scenario**

Exception is causing crashes on customer machines.

**Bugs this fixes:**

https://devdiv.visualstudio.com/DevDiv/_workitems?id=403647

**Workarounds, if any**

This is an IOException, so it's usually caused by the output path being non-writable for some reason. The user could choose a different output path which may be writable.

**Risk**

Very low risk -- simply catches exception and propagates the failure upward.

**Performance impact**

None -- additional try/catch.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

This code path was only used by portable PDBs, so it's likely that users weren't using portable PDBs as often and thus wouldn't hit the crash as often if an I/O failure occurred.

**How was the bug found?**

Watson.

### CODE REVIEW

**Comment by agocke** [comment]
cc @jaredpar @dotnet/roslyn-compiler

File: src/Compilers/Core/Portable/PEWriter/PeWriter.cs [hunk 1]
```diff
@@ -154,7 +154,14 @@ internal static class PeWriter
                     Stream portablePdbStream = getPortablePdbStreamOpt();
                     if (portablePdbStream != null)
                     {
-                        portablePdbBlob.WriteContentTo(portablePdbStream);
+                        try
+                        {
+                            portablePdbBlob.WriteContentTo(portablePdbStream);
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/PEWriter/PeWriter.cs]: Unit tests?

**Comment by tmat** [comment]
👍

**Comment by agocke** [comment]
Woops, forgot to stage the test change. Thanks!

**Comment by agocke** [comment]
@MeiChin-Tsai for ask mode approval

**Comment by MeiChin-Tsai** [comment]
Approved. Please make sure all checks are successful. thx.

**Comment by jaredpar** [comment]
retest windows_eta_open_prtest please

**Comment by agocke** [comment]
retest windows_eta_open_prtest please

original failed due to "NuGet cache corruption with null files"

**Comment by jaredpar** [comment]
retest windows_eta_open_prtest please

**Comment by agocke** [comment]
retest windows_eta_open_prtest please

more nuget corruption

**Pull request merged**